service/src/test/java/org/sunbird/learner/actors/UserManagementActorTestExtended.java

```diff
@@ -0,0 +1,349 @@
+  }
+
+  @Test
+  public void testUpdateUserFrameworkFailureInvalidGradeLevel() {
+    Request reqObj = getRequest(false, false, true, false, false);
+    ProjectCommonException res = doUpdateActorCallFailure(reqObj);
+    assertTrue(res.getCode().equals(ResponseCode.invalidParameterValue.getErrorCode()));
+  }
+
+  @Test
+  public void testUpdateUserFrameworkFailureInvalidMedium() {
+    Request reqObj = getRequest(false, false, false, true, false);
+    ProjectCommonException res = doUpdateActorCallFailure(reqObj);
+    assertTrue(res.getCode().equals(ResponseCode.invalidParameterValue.getErrorCode()));
+  }
+
+  @Test
+  public void testUpdateUserFrameworkFailureInvalidBoard() {
+    Request reqObj = getRequest(false, false, false, false, true);
+    ProjectCommonException res = doUpdateActorCallFailure(reqObj);
+    assertTrue(res.getCode().equals(ResponseCode.invalidParameterValue.getErrorCode()));
+  }
+
+  @Test
+  public void testUpdateUserFrameworkFailureInvalidFrameworkId() {
+    Request reqObj = getRequest(false, true, false, false, false);
+    ProjectCommonException res = doUpdateActorCallFailure(reqObj);
+    assertTrue(res.getCode().equals(ResponseCode.errorNoFrameworkFound.getErrorCode()));
+  }
+
+  private Response doUpdateActorCallSuccess(Request reqObj) {
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+    mockUserServiceForValidatingUserId(reqObj);
+    mockuserExtDao(reqObj);
+    subject.tell(reqObj, probe.getRef());
+    Response res = probe.expectMsgClass(duration("200 second"), Response.class);
+    return res;
+  }
+
+  private ProjectCommonException doUpdateActorCallFailure(Request reqObj) {
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+    mockUserServiceForValidatingUserId(reqObj);
+    mockuserExtDao(reqObj);
+    subject.tell(reqObj, probe.getRef());
+    ProjectCommonException res =
+        probe.expectMsgClass(duration("200 second"), ProjectCommonException.class);
+    return res;
+  }
+
+  private Request getRequest(
+      boolean success, boolean id_, boolean grade, boolean medium, boolean board) {
+    Request reqObj = new Request();
+    reqObj.setOperation(ActorOperations.UPDATE_USER.getValue());
+    Map<String, Object> innerMap = new HashMap<>();
+    innerMap.put(JsonKey.ID, userId);
+    Map<String, Object> frameworkMap = getFrameworkDetails(success, id_, grade, medium, board);
+
+    innerMap.put(JsonKey.FRAMEWORK, frameworkMap);
+    Map<String, Object> request = new HashMap<String, Object>();
+    request.put(JsonKey.USER, innerMap);
+    request.put(JsonKey.USER_ID, userId);
+    request.put(JsonKey.FRAMEWORK, frameworkMap);
+    reqObj.setRequest(request);
+    Map<String, Object> context = new HashMap<>();
+    context.put(JsonKey.REQUESTED_BY, "someValue");
+    context.put(JsonKey.USER_ID, userId);
+    reqObj.setContext(context);
+    return reqObj;
+  }
+
+  private Map<String, Object> getFrameworkDetails(
```

Comment:
Rewrite this function -> pass key/value from outside - which you want to over-ride.

No need to add so many boolean.